### PR TITLE
fix: [CDS-68578]: return winrm command result with error message

### PR DESCRIPTION
**Pull request by hingerabhinav**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/47709)

### CODE REVIEW

**Comment by yogesh-chauhan** [comment]
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/winrm/WinRmSession.java` line 166 at r1](https://reviewable.io/reviews/harness/harness-core/47709#-NVUP4j9-3XTTp6J9TAW:-NVUP4j9-3XTTp6J9TAX:b2ybnk3) ([raw file](https://github.com/harness/harness-core/blob/bb524a0351c2f96c4202486c99786c627fe68304/930-delegate-tasks/src/main/java/io/harness/delegate/task/winrm/WinRmSession.java#L166)):*
> ```Java
> 
>       if (winRmCommandResult != null && winRmCommandResult.isSuccess()) {
>         return 1;
> ```

shouldn't success be 0 exit code?

**Review by vaibhav-si — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/47709)* status: Some of the checks are too old. 3 file(s) to review. 0 of 2 approvals obtained. 

___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/winrm/WinRmSession.java` line 166 at r1](https://reviewable.io/reviews/harness/harness-core/47709#-NVUP4j9-3XTTp6J9TAW:-NVY1oB6CMNgpkpYvGdf:b32v57) ([raw file](https://github.com/harness/harness-core/blob/bb524a0351c2f96c4202486c99786c627fe68304/930-delegate-tasks/src/main/java/io/harness/delegate/task/winrm/WinRmSession.java#L166)):*
<details><summary><i>Previously, yogesh-chauhan (Yogesh Chauhan) wrote…</i></summary><blockquote>

shouldn't success be 0 exit code?
</blockquote></details>

+1

File: 930-delegate-tasks/src/main/java/io/harness/delegate/task/winrm/WinRmSession.java [hunk 1]
```diff
@@ -156,13 +157,16 @@ public int executeCommandWithKerberos(String command, Writer output, boolean isO
...
-                 logCallback, output, isOutputWriter, args.getEnvironmentMap())
-          ? 0
-          : 1;
+      WinRmCommandResult winRmCommandResult = SshHelperUtils.executeLocalCommand(
+          format(COMMAND_PLACEHOLDER, InstallUtils.getPath(ClientTool.HARNESS_PYWINRM, HarnessPywinrmVersion.V0_4),
+              args.getArgs(commandFile.getAbsolutePath())),
+          logCallback, output, isOutputWriter, args.getEnvironmentMap());
+
+      if (winRmCommandResult != null && winRmCommandResult.isSuccess()) {
+        return 1;
```
Comment (yogesh-chauhan) [inline 930-delegate-tasks/src/main/java/io/harness/delegate/task/winrm/WinRmSession.java:166]: this should be return 0

**Review by yogesh-chauhan — CHANGES_REQUESTED** [review]

**Review by hingerabhinav — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/47709)* status: 5 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`930-delegate-tasks/src/main/java/io/harness/delegate/task/winrm/WinRmSession.java` line 166 at r1](https://reviewable.io/reviews/harness/harness-core/47709#gh-1194648182:-NVYIAnW4hrbFJi1u32B:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/bb524a0351c2f96c4202486c99786c627fe68304/930-delegate-tasks/src/main/java/io/harness/delegate/task/winrm/WinRmSession.java#L166)):*
<details><summary><i>Previously, yogesh-chauhan (Yogesh Chauhan) wrote…</i></summary><blockquote>

this should be return 0
</blockquote></details>

Done.

**Comment by hingerabhinav** [comment]
trigger uts

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 2 of 3 files at r1, 3 of 3 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/47709)* status: 5 file(s) to review. 1 of 2 approvals obtained.

**Review by vaibhav-si — APPROVED** [review]
> Reviewed 2 of 3 files at r1, 3 of 3 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/47709)* status: :shipit: complete! ✓

**Comment by hingerabhinav** [comment]
trigger review

**Pull request merged**